pkg/cloudprovider/types.go

```diff
@@ -245,8 +245,8 @@ func (ofs Offerings) Available() Offerings {
 // Compatible returns the offerings based on the passed requirements
 func (ofs Offerings) Compatible(reqs scheduling.Requirements) Offerings {
```

Comment:
It could be but then the concepts of Zone and CapacityType are no longer special and I think we want to keep them that way. This also forces other cloudproviders to think about constraints besides those two.

What this does is that if there are additional constraints specific to a cloudprovider then they can handle it within their implementation when initially creating offerings like at https://github.com/aws/karpenter-provider-aws/pull/6216/files#diff-b0143481b3c25f93f1ad9de2b0a7b67dbbe61c01c4187ade18c26759a0ff5c04R343

WDYT?

Reply:
The problem is that we do need to consider these additional constraints in the upstream code. Consider a case where we have a `NotIn` requirement for a zone-id. When we're computing consolidation decisions and filtering for offerings, we would still include the offering with that zone-id since we're only filtering on zone. This could lead to us making incorrect decisions around pricing.